src/EditorFeatures/Core.Wpf/InlineRename/AbstractEditorInlineRenameService.cs

```diff
@@ -194,21 +177,10 @@ private IInlineRenameInfo GetRenameInfo(Document document, int position, Cancell
                         {
                             return new FailureInlineRenameInfo(EditorFeaturesResources.You_cannot_rename_elements_from_previous_submissions);
                         }
-                    }
-                    else
-                    {
-                        var sourceText = location.SourceTree.GetTextAsync(cancellationToken).WaitAndGetResult(cancellationToken);
-                        var textSnapshot = sourceText.FindCorrespondingEditorTextSnapshot();
-
-                        if (textSnapshot != null)
+                        else
                         {
-                            var buffer = textSnapshot.TextBuffer;
-                            var originalSpan = location.SourceSpan.ToSnapshotSpan(textSnapshot).TranslateTo(buffer.CurrentSnapshot, SpanTrackingMode.EdgeInclusive);
-
-                            if (buffer.IsReadOnly(originalSpan) || !navigationService.CanNavigateToSpan(workspace, document.Id, location.SourceSpan))
-                            {
-                                return new FailureInlineRenameInfo(EditorFeaturesResources.You_cannot_rename_this_element);
-                            }
+                            // We eventually need to return the symbol locations, so we must convert each location to a DocumentSpan since our return type is language-agnostic.
+                            documentSpans.Add(new DocumentSpan(document, location.SourceSpan));
```

Comment:
this is nested wrong.  you are *only* doing this in the submision case.  but the previous code only did it in the *non-submission* case.

Reply:
You're right, thank you for the catch!